massa-execution-worker/src/execution.rs

```diff
@@ -419,6 +419,18 @@ impl ExecutionState {
         // acquire write access to the context
         let mut context = context_guard!(self);
 
+        let de_slot = denunciation.get_slot();
+
+        let last_start_period = self.final_state.read().last_start_period;
```

Comment:
last_start_period is a config parameter, it can be passed directly in the Execution config to avoid having to fetch it from final_state with a read-lock